src/client/Client.ts

```diff
@@ -8,6 +8,10 @@ import { ClientOptionsDefaults } from '../util/Constants';
 import type { Store } from '../lib/structures/base/Store';
 import type { Piece } from '../lib/structures/base/Piece';
 
+export interface Plugin {
+	[Client.plugin]: Function;
```

Comment:
Can't we type the function? (might me able to make it know that `this` is the Client instance

Reply:
The answer was: Unnecessary.